### ISSUE TITLE
Consolidate and write explanation of passable types representation invariants

### ISSUE THREAD

**Issue by erights** [Agoric/Agoric#6148]
As requested at https://github.com/Agoric/agoric-sdk/pull/6057#discussion_r963157976

Our passable types come in two levels of abstraction -- the [@endo/marshal](https://github.com/endojs/endo/tree/master/packages/marshal) level, characterized by `passStyleOf`.

One marshal passStyle is "tagged", which is the escape hatch used by [@endo/patterns](https://github.com/endojs/endo/tree/master/packages/patterns) and [@agoric/store](https://github.com/Agoric/agoric-sdk/tree/master/packages/store) to build and encode higher level data types, such as CopySets, CopyBags, CopyMaps, and the various Matchers for pattern matching.

At this @agoric/store level of abstraction, some tagged object are recognized: 
   * They are already known to be a proper tagged record by the checking done by `passStyleOf`.
   * Their tag value is recognized by @agoric/store
   * Their representation passes all the invariants that @agoric/store insists on.

Any tagged record recognized by @endo/marshal but not by @agoric/store is still a valid Passable. It may be recognized by something else. A system that does not recognize it may still pass it opaquely to something that might understand it.

The encoding of the recognized @agoric/store types into tagged record must satisfy an amazing constraint on ordering.
   * The marshal level defined a rank-full-order among all passable suitable for sorting. These rules therefore give us a rank order among taggeds that cannot take into account their meaning at the store level.
   * The store level defines a partial order among keys, which include many tagged records. If keyCompare(x, y) < 0, then rankCompare(x, y) <= 0. 

The purpose of the key partial order is to enable inequality comparisons meaningful at the store level of abstraction. For example, if x and y are sets, then keyCompare(x, y) < 0 means that x is a proper subset of y. The constraint means that x must also not have a higher rank than y in the rank full order.

For both the @endo/marshal and the @agoric/store levels, we need a self-contained explanation of the recognized tag records and what their representational invariants are. This should consolidate material that is currently scattered elsewhere, including
   * https://github.com/endojs/endo/blob/master/packages/marshal/README.md
   * https://github.com/endojs/endo/blob/master/packages/marshal/src/types.js
   * https://github.com/Agoric/agoric-sdk/pull/4435/files
   * ...What else? I have not been able to find some important explanations that we put effort into

**Comment by erights** [comment]
Hi @Tyrosine22 I just removed ChrisB and added you. It would be cool to get this written down in a consolidated and clear manner.

**Comment by erights** [comment]
@kriskowal @gibson042 , do you agree that this is now adequately addressed at least by ocapn docs, so we can close this?

**Comment by kriskowal** [comment]
I don’t think we should rely on the OCapN docs to be a source of truth for Endo’s types at this time.
